From 3d86ad9798eefe396dab77a4b4aa9709d3d00fbd Mon Sep 17 00:00:00 2001
From: Mustafa Ali <quic_mustali@quicinc.com>
Date: Thu, 6 Mar 2025 11:16:44 +0200
Subject: [PATCH] gstreamer1.0-plugins-bad: Add meson option to build all
 plugins

- Default value is to build all plugins
- If reduced set of plugins is selected, then only are build:
    gst-libs/gst/wayland
    ext/wayland

Upstream-Status: Inappropriate [Qualcomm specific]

Signed-off-by: Mustafa Ali <quic_mustali@quicinc.com>
---
 ext/meson.build          | 172 ++++++++++++++++++++-------------------
 gst-libs/gst/meson.build |  60 +++++++-------
 meson.build              |  32 +++++---
 meson_options.txt        |   4 +
 4 files changed, 144 insertions(+), 124 deletions(-)

diff --git a/ext/meson.build b/ext/meson.build
index 9fc71a9..b7aed6c 100644
--- a/ext/meson.build
+++ b/ext/meson.build
@@ -1,84 +1,88 @@
-subdir('aes')
-subdir('analyticsoverlay')
-subdir('assrender')
-subdir('aom')
-subdir('avtp')
-subdir('bs2b')
-subdir('bz2')
-subdir('chromaprint')
-subdir('closedcaption')
-subdir('codec2json')
-subdir('colormanagement')
-subdir('curl')
-subdir('dash')
-subdir('dc1394')
-subdir('directfb')
-subdir('dtls')
-subdir('dts')
-subdir('faac')
-subdir('faad')
-subdir('fdkaac')
-subdir('flite')
-subdir('fluidsynth')
-subdir('gme')
-subdir('gs')
-subdir('gsm')
-subdir('gtk')
-subdir('hls')
-subdir('iqa')
-subdir('isac')
-subdir('ladspa')
-subdir('lc3')
-subdir('lcevcdecoder')
-subdir('lcevcencoder')
-subdir('ldac')
-subdir('libde265')
-subdir('lv2')
-subdir('mdns')
-subdir('modplug')
-subdir('mpeg2enc')
-subdir('mplex')
-subdir('musepack')
-subdir('neon')
-subdir('nvcomp')
-subdir('nvdswrapper')
-subdir('onnx')
-subdir('openal')
-subdir('openaptx')
-subdir('opencv')
-subdir('openexr')
-subdir('openh264')
-subdir('openjpeg')
-subdir('openmpt')
-subdir('openni2')
-subdir('opus')
-subdir('qroverlay')
-subdir('qt6d3d11')
-subdir('resindvd')
-subdir('rsvg')
-subdir('rtmp')
-subdir('sbc')
-subdir('sctp')
-subdir('smoothstreaming')
-subdir('sndfile')
-subdir('soundtouch')
-subdir('spandsp')
-subdir('srt')
-subdir('srtp')
-subdir('svtav1')
-subdir('svthevcenc')
-subdir('svtjpegxs')
-subdir('teletextdec')
-subdir('ttml')
-subdir('voaacenc')
-subdir('voamrwbenc')
-subdir('vulkan')
-subdir('wayland')
-subdir('webrtc')
-subdir('webrtcdsp')
-subdir('webp')
-subdir('wildmidi')
-subdir('wpe')
-subdir('x265')
-subdir('zxing')
-subdir('zbar')
+if build_all_plugins
+  subdir('aes')
+  subdir('analyticsoverlay')
+  subdir('assrender')
+  subdir('aom')
+  subdir('avtp')
+  subdir('bs2b')
+  subdir('bz2')
+  subdir('chromaprint')
+  subdir('closedcaption')
+  subdir('codec2json')
+  subdir('colormanagement')
+  subdir('curl')
+  subdir('dash')
+  subdir('dc1394')
+  subdir('directfb')
+  subdir('dtls')
+  subdir('dts')
+  subdir('faac')
+  subdir('faad')
+  subdir('fdkaac')
+  subdir('flite')
+  subdir('fluidsynth')
+  subdir('gme')
+  subdir('gs')
+  subdir('gsm')
+  subdir('gtk')
+  subdir('hls')
+  subdir('iqa')
+  subdir('isac')
+  subdir('ladspa')
+  subdir('lc3')
+  subdir('lcevcdecoder')
+  subdir('lcevcencoder')
+  subdir('ldac')
+  subdir('libde265')
+  subdir('lv2')
+  subdir('mdns')
+  subdir('modplug')
+  subdir('mpeg2enc')
+  subdir('mplex')
+  subdir('musepack')
+  subdir('neon')
+  subdir('nvcomp')
+  subdir('nvdswrapper')
+  subdir('onnx')
+  subdir('openal')
+  subdir('openaptx')
+  subdir('opencv')
+  subdir('openexr')
+  subdir('openh264')
+  subdir('openjpeg')
+  subdir('openmpt')
+  subdir('openni2')
+  subdir('opus')
+  subdir('qroverlay')
+  subdir('qt6d3d11')
+  subdir('resindvd')
+  subdir('rsvg')
+  subdir('rtmp')
+  subdir('sbc')
+  subdir('sctp')
+  subdir('smoothstreaming')
+  subdir('sndfile')
+  subdir('soundtouch')
+  subdir('spandsp')
+  subdir('srt')
+  subdir('srtp')
+  subdir('svtav1')
+  subdir('svthevcenc')
+  subdir('svtjpegxs')
+  subdir('teletextdec')
+  subdir('ttml')
+  subdir('voaacenc')
+  subdir('voamrwbenc')
+  subdir('vulkan')
+  subdir('wayland')
+  subdir('webrtc')
+  subdir('webrtcdsp')
+  subdir('webp')
+  subdir('wildmidi')
+  subdir('wpe')
+  subdir('x265')
+  subdir('zxing')
+  subdir('zbar')
+else
+  subdir('wayland')
+endif
diff --git a/gst-libs/gst/meson.build b/gst-libs/gst/meson.build
index 5f51ceb..f75af54 100644
--- a/gst-libs/gst/meson.build
+++ b/gst-libs/gst/meson.build
@@ -1,29 +1,33 @@
-subdir('uridownloader')
+if build_all_plugins
+  subdir('uridownloader')
 
-subdir('adaptivedemux')
-subdir('audio')
-subdir('basecamerabinsrc')
-subdir('codecparsers')
-subdir('codecs')
-subdir('d3dshader')
-subdir('d3d11')
-subdir('d3d12')
-# cuda can depend on d3d11
-subdir('cuda')
-subdir('dxva')
-subdir('insertbin')
-subdir('interfaces')
-subdir('isoff')
-subdir('mpegts')
-subdir('mse')
-subdir('opencv')
-subdir('play')
-subdir('player')
-subdir('sctp')
-subdir('transcoder')
-subdir('va')
-subdir('vulkan')
-subdir('wayland')
-subdir('webrtc')
-subdir('winrt')
-subdir('analytics')
+  subdir('adaptivedemux')
+  subdir('audio')
+  subdir('basecamerabinsrc')
+  subdir('codecparsers')
+  subdir('codecs')
+  subdir('d3dshader')
+  subdir('d3d11')
+  subdir('d3d12')
+  # cuda can depend on d3d11
+  subdir('cuda')
+  subdir('dxva')
+  subdir('insertbin')
+  subdir('interfaces')
+  subdir('isoff')
+  subdir('mpegts')
+  subdir('mse')
+  subdir('opencv')
+  subdir('play')
+  subdir('player')
+  subdir('sctp')
+  subdir('transcoder')
+  subdir('va')
+  subdir('vulkan')
+  subdir('wayland')
+  subdir('webrtc')
+  subdir('winrt')
+  subdir('analytics')
+else
+  subdir('wayland')
+endif
diff --git a/meson.build b/meson.build
index 5c983c5..5327f77 100644
--- a/meson.build
+++ b/meson.build
@@ -583,13 +583,18 @@ else
   pathsep = ':'
 endif
 
+build_all_plugins = get_option('build-all-plugins')
+
 subdir('gst-libs')
-subdir('gst')
-subdir('sys')
-subdir('ext')
-subdir('tests')
-subdir('data')
-subdir('tools')
+
+if build_all_plugins
+  subdir('gst')
+  subdir('sys')
+  subdir('ext')
+  subdir('tests')
+  subdir('data')
+  subdir('tools')
+endif
 
 if have_orcc
   update_orc_dist_files = find_program('scripts/update-orc-dist-files.py')
@@ -615,13 +620,14 @@ if have_orcc
 endif
 
 # xgettext is optional (on Windows for instance)
-if find_program('xgettext', required : get_option('nls')).found()
-  cdata.set('ENABLE_NLS', 1)
-  subdir('po')
+if build_all_plugins
+  if find_program('xgettext', required : get_option('nls')).found()
+    cdata.set('ENABLE_NLS', 1)
+    subdir('po')
+  endif
+  subdir('scripts')
 endif
 
-subdir('scripts')
-
 # Set release date
 if gst_version_nano == 0
   extract_release_date = find_program('scripts/extract-release-date-from-doap-file.py')
@@ -639,7 +645,9 @@ configure_file(output : 'config.h', configuration : cdata)
 
 meson.add_dist_script('scripts/gen-changelog.py', meson.project_name(), '1.24.0', meson.project_version())
 
-subdir('docs')
+if build_all_plugins
+  subdir('docs')
+endif
 
 plugin_names = []
 gst_plugins = []
diff --git a/meson_options.txt b/meson_options.txt
index 35e252d..090ea5d 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -299,6 +299,10 @@ option('glib_checks', type : 'boolean', value : true, yield : true, description
   deprecated: {'enabled' : 'true', 'disabled' : 'false', 'auto' : 'false'},
 )
 
+# Build options
+option('build-all-plugins', type : 'boolean', yield : true,
+       description : 'Enable build of all plugins')
+
 # Deprecated, kept for backward compat
 option('gobject-cast-checks', type : 'feature', value : 'auto', yield : true,
        description: 'Enable run-time GObject cast checks (auto = enabled for development, disabled for stable releases)',
-- 
2.25.1

